src/test/java/com/iexec/sms/tee/session/palaemon/PalaemonSessionServiceTests.java

```diff
@@ -264,7 +263,10 @@ void shouldGetPalaemonTokensWithoutAppComputeSecret() {
                         IexecEnvUtils.IEXEC_INPUT_FILE_NAME_PREFIX + "2", "file2"));
         assertThat(tokens)
                 .containsEntry(IEXEC_APP_DEVELOPER_SECRET_0, "");
-        assertThat(tokens.get(REQUESTER_SECRETS)).isEqualTo(Collections.emptyMap());
+        assertThat(tokens.get(REQUESTER_SECRETS)).isEqualTo(Map.of(
```

Comment:
Why isn't it emplty anymore?

Reply:
As the test is designed, requester secrets in deal parameters are not empty.
Previously, as the max requester secret count was not defined, there was a barrier and map was empty.

Now and following yesterday discussion, the deal parameter should be considered consistent and as being validated previously in the process.

Maybe the test can be split in 2 tests:
1. If deal parameters have secrets which does not exist, it produces the map with empty values (see https://github.com/iExecBlockchainComputing/iexec-sms/blob/feature/remove-checks-on-max-requester-secrets-count/src/main/java/com/iexec/sms/tee/session/palaemon/PalaemonSessionService.java#L272)
2. if deal parameters do not have secrets, the map should be empty (we need a test without calling this method https://github.com/iExecBlockchainComputing/iexec-sms/blob/feature/remove-checks-on-max-requester-secrets-count/src/test/java/com/iexec/sms/tee/session/palaemon/PalaemonSessionServiceTests.java#L409)

What do you think ?

Reply:
Thanks for the explanation.
I do agree on what you suggest, splitting the test seems good to me.

Reply:
Tests reworked a little bit in 0c43867